lib/improver/tests/utilities/test_statistical_operations.py

```diff
@@ -0,0 +1,278 @@
+    cubes above.
+    """
+    reference_array = np.zeros(shape=(16,))
+    reference_array[:4] = 0.
+    reference_array[4:12] = np.linspace(0., 0.875, 8)
+    reference_array[12:] = 1.
+    return reference_array.reshape(4, 4)
+
+
+class Test__init__(IrisTest):
+    """ Test initialisation of the ProbabilitiesFromPercentiles2D class """
+    def setUp(self):
+        """ Set up test cube """
+        self.test_cube = set_up_percentiles_cube()
+        self.new_name = "probability"
+
+    def test_basic(self):
+        """ Test name and default ordering are correctly set """
+        pfp_instance = ProbabilitiesFromPercentiles2D(self.test_cube,
+                                                      self.new_name)
+        self.assertEqual(pfp_instance.output_name, self.new_name)
+
+    def test_naming(self):
+        """ Test default naming """
+        pfp_instance = ProbabilitiesFromPercentiles2D(self.test_cube)
+        self.assertEqual(pfp_instance.output_name,
+                         "probability_of_{}".format(self.test_cube.name()))
+
+
+class Test__repr__(IrisTest):
+    """ Test string representation """
+    def test_basic(self):
+        """ Compare __repr__ string with expectation """
+        new_name = "probability"
+        test_cube = set_up_percentiles_cube()
+        expected = ('<ProbabilitiesFromPercentiles2D: percentiles_'
+                    'cube: {}, output_name: {}'.format(
+                        test_cube, new_name))
+        result = str(ProbabilitiesFromPercentiles2D(test_cube,
+                                                    new_name))
+        self.assertEqual(result, expected)
+
+
+class Test_create_probability_cube(IrisTest):
+    """
+    Test creation of new probability cube based on percentile and orography
+    input cubes
+    """
+    def setUp(self):
+        """ Set up a probability cube from percentiles and orography """
+        self.percentiles_cube = set_up_percentiles_cube()
+        self.percentile_coordinate = find_percentile_coordinate(
+            self.percentiles_cube)
+        self.new_name = "probability"
+        self.pfp_instance = ProbabilitiesFromPercentiles2D(
+            self.percentiles_cube, self.new_name)
+        self.orography_cube = set_up_threshold_cube()
+
+    def test_attributes(self):
+        """ Test name and units are correctly set """
+        result = self.pfp_instance.create_probability_cube(
+            self.percentiles_cube)
+        self.assertEqual(result.units, "1")
+        self.assertEqual(result.name(), self.new_name)
+
+    def test_coordinate_collapse(self):
+        """ Test probability cube has no percentile coordinate """
+        result = self.pfp_instance.create_probability_cube(
+            self.percentiles_cube)
+        with self.assertRaises(CoordinateNotFoundError):
+            find_percentile_coordinate(result)
+
+
+class Test_percentile_interpolation(IrisTest):
```

Comment:
I'm not sure how far we should go with looking at edge cases, but there could be a few documented in the unit tests. 
One that could be interesting is when you only have one percentile in your input cube.
There's probably others.

Reply:
I have raised an exception for this to the code and an associated test. We can't do anything useful with a single valued percentile coordinate.